[BACKPORT] stm32f7:Add Serial Tx DMA

davids5 opened the pull request:
[BACKPORT] arch/arm/src/stm32f0l0g0/stm32_serial_v1.c:  CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO

[BACKPORT] arch/arm/src/stm32l4/stm32l4_serial.c: CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32f0l0g0/stm32_serial_v1.c: SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32l4/stm32l4_serial.c: SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32/stm32_hciuart.c:  CONFIG_STM32_HCIUART_DMAPRIO->CONFIG_STM32_HCIUART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32/stm32_serial.c:  SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  Add Tx U[S]ART DMA.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  Serial nxsyle fixes (sans long table lines).

[BACKPORT] Merged in david_s5/nuttx-5/David-Sidrane/archarmsrcstm32f7stm32_serialc-fix-typo-1575381167793 (pull request #1087)

arch/arm/src/stm32f7/stm32_serial.c:  Fix typo in UART7

Approved-by: Gregory Nutt <gnutt@nuttx.org>

NaterGator commented on nuttx/fs/fat/fs_fat32.c:
```diff
@@ -962,6 +962,14 @@ static off_t fat_seek(FAR struct file *filep, off_t offset, int whence)
 
+  /* Special case for ftell */
+
+  if (offset == 0 && whence == SEEK_CUR) {
```
For reviewers: this condition comes from [this](https://github.com/PX4/NuttX/blob/f0f4bdc872d324f64e9d93d6f8989d3c1dfa2633/nuttx/libc/stdio/lib_ftell.c#L118) implementation.

LorenzMeier commented:
@nuttx Is this a change you're willing to take upstream? Note that we're going to move to current NuttX in one of the next two releases, so we're committed to get back in sync with upstream and having this fix would be important. Thanks!

LorenzMeier commented:
I'm going to merge it here as a hotfix for the NuttX snapshot we're currently using.

nuttx commented:
I understand the purpose of the change, but it does not seem correct.  First of all, it adds a left brace, but no right brace.  Perhaps that is just missing in the snippet.

nuttx commented:
Okay.  The return value is incorrect.  The seek() method returns 0 on success and a negative value on value.  On success, the implementation of lseek() in fs/vfs/fs_lseek will return filep->f_pos;

nuttx commented:
I committed a functionally equivalent change that also catches a couple of other use cases where we may attempt to seek to the current position:  https://bitbucket.org/nuttx/nuttx/commits/610afe9cd9440d77dc95ab253fcf9db9be1a05a9

NaterGator commented:
Thanks @nuttx. @LorenzMeier we should definitely get the correct return value for fseek in.

nuttx commented:
Returning the offset which is type off_t will typically work as well, as least on a 32-bit ARM.  An off_t is a normally an unsigned long value and the same size as the int return value.  The pass/fail test is that the return value is < 0 or >= 0.  So for offsets less then 2^31 everything is okay.

Problems would occur only for large files > 2Gb, if large files are supported when off_t because 64  bits, or for other architectures where sizeof(int) < sizeof(off_t).

davids5 committed:
[BACKPORT] stm32f7:Add Serial Tx DMA

[BACKPORT] arch/arm/src/stm32f0l0g0/stm32_serial_v1.c:  CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO

[BACKPORT] arch/arm/src/stm32l4/stm32l4_serial.c: CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32f0l0g0/stm32_serial_v1.c: SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32l4/stm32l4_serial.c: SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32/stm32_hciuart.c:  CONFIG_STM32_HCIUART_DMAPRIO->CONFIG_STM32_HCIUART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32/stm32_serial.c:  SERIAL_HAVE_DMA->SERIAL_HAVE_RXDMA.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  CONFIG_USART_DMAPRIO->CONFIG_USART_RXDMAPRIO.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  Add Tx U[S]ART DMA.

[BACKPORT] arch/arm/src/stm32f7/stm32_serial.c:  Serial nxsyle fixes (sans long table lines).

[BACKPORT] Merged in david_s5/nuttx-5/David-Sidrane/archarmsrcstm32f7stm32_serialc-fix-typo-1575381167793 (pull request #1087)

arch/arm/src/stm32f7/stm32_serial.c:  Fix typo in UART7

Approved-by: Gregory Nutt <gnutt@nuttx.org>
